src/core/base.jl

```diff
@@ -40,24 +40,43 @@ InfrastructureModels.@def im_fields begin
 end
 
 # default generic constructor
-function InitializeInfrastructureModel(InfrastructureModel::Type, data::Dict{String,<:Any}, global_keys::Set{String}; ext = Dict{Symbol,Any}(), setting = Dict{String,Any}(), jump_model::JuMP.AbstractModel=JuMP.Model())
+function InitializeInfrastructureModel(
+    InfrastructureModel::Type, data::Dict{String, <:Any}, global_keys::Set{String};
+    ext = Dict{Symbol, Any}(), setting = Dict{String, Any}(),
+    jump_model::JuMP.AbstractModel = JuMP.Model(), default_it::Symbol = :it)
     @assert InfrastructureModel <: AbstractInfrastructureModel
 
-    ref = ref_initialize(data, global_keys) # reference data
-
-    var = Dict{Symbol,Any}(:nw => Dict{Int,Any}())
-    con = Dict{Symbol,Any}(:nw => Dict{Int,Any}())
-    sol = Dict{Symbol,Any}(:nw => Dict{Int,Any}())
-    sol_proc = Dict{Symbol,Any}(:nw => Dict{Int,Any}())
+    cnw = Inf # Current network index (used as default multinetwork index).
+    ref = Dict{Symbol, Any}(:it => Dict{Symbol, Any}())
+    var = Dict{Symbol, Any}(:it => Dict{Symbol, Any}())
+    con = Dict{Symbol, Any}(:it => Dict{Symbol, Any}())
+    sol = Dict{Symbol, Any}(:it => Dict{Symbol, Any}())
+    sol_proc = Dict{Symbol, Any}(:it => Dict{Symbol, Any}())
+
+    function initialize_it(it::Symbol, global_keys::Set{String} = Set{String}())
+        ref[:it][it] = ref_initialize(data, string(it), global_keys)
+        var[:it][it] = Dict{Symbol, Any}(:nw => Dict{Int, Any}())
+        con[:it][it] = Dict{Symbol, Any}(:nw => Dict{Int, Any}())
+        sol[:it][it] = Dict{Symbol, Any}(:nw => Dict{Int, Any}())
+        sol_proc[:it][it] = Dict{Symbol, Any}(:nw => Dict{Int, Any}())
+
+        for (nw_id, nw) in ref[:it][it][:nw]
+            var[:it][it][:nw][nw_id] = Dict{Symbol, Any}()
+            con[:it][it][:nw][nw_id] = Dict{Symbol, Any}()
+            sol[:it][it][:nw][nw_id] = Dict{Symbol, Any}()
+            sol_proc[:it][it][:nw][nw_id] = Dict{Symbol, Any}()
+        end
 
-    for (nw_id, nw) in ref[:nw]
-        nw_var = var[:nw][nw_id] = Dict{Symbol,Any}()
-        nw_con = con[:nw][nw_id] = Dict{Symbol,Any}()
-        nw_sol = sol[:nw][nw_id] = Dict{Symbol,Any}()
-        nw_sol_proc = sol_proc[:nw][nw_id] = Dict{Symbol,Any}()
+        cnw = min(cnw, minimum([k for k in keys(var[:it][it][:nw])]))
     end
 
-    cnw = minimum([k for k in keys(var[:nw])])
+    if ismultiinfrastructure(data)
+        its = [Symbol(key) for key in keys(data["it"])]
+        initialize_it.(its, Ref(global_keys))
+        ref[:link_component] = ref_initialize(data, global_keys)
```

Comment:
Here, `link_component` is one level above the infrastructure-specific components (e.g., `delivery_gen` in GasPowerModels). I think having this upper-level key for linking components makes sense to avoid the proliferation of new multi-infrastructure component types at the top level of the dictionary. We can debate this on Thursday.

Reply:
We will create a top-level key called something like `interdependency`, below this a possible `nw` key, and below this the components that would previously have gone in `link_component`.